src/url.cc

```diff
@@ -536,16 +536,15 @@ URL::absolute() const
  *        After copying it on in the first place! Would be less code to merge the two with a flag parameter.
  *        and never copy the query-string part in the first place
  */
+
 char *
-urlCanonicalClean(const HttpRequest * request)
+urlCanonicalClean(const SBuf &url, const bool stripQuery)
 {
     LOCAL_ARRAY(char, buf, MAX_URL);
-
-    snprintf(buf, sizeof(buf), SQUIDSBUFPH, SQUIDSBUFPRINT(request->effectiveRequestUri()));
+    snprintf(buf, sizeof(buf), SQUIDSBUFPH, SQUIDSBUFPRINT(url));
     buf[sizeof(buf)-1] = '\0';
 
-    // URN, CONNECT method, and non-stripped URIs can go straight out
-    if (Config.onoff.strip_query_terms && !(request->method == Http::METHOD_CONNECT || request->url.getScheme() == AnyP::PROTO_URN)) {
+    if (stripQuery) {
```

Comment:
As we discussed during the conference call, please see whether it makes sense to pass the method and the scheme to this function instead of the boolean stripQuery. The function itself will (continue to) check Config.onoff.strip_query_terms and CONNECT/URN exceptions in that case, of course.